### PR TITLE
Feat: Improved SSL generation

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

- If APP_DOMAIN is set, only allow certificates for this domain
- If APP_DOMAIN is not set, allow the first one, and remember it's the main domain. This makes DO 1click frictionless
- Renamed 'master' domain to 'main' domain in logs

---

- Custom domains are not affected. SSL generation for these is triggered from Console (you click to trigger it 'Verify now')
- Renewing is not affected. Renewing is done with a schedule in Resque.

## Test Plan

- [x] Manual QA on DO:

CleanShot 2022-03-29 at 12 16 45
CleanShot 2022-03-29 at 12 16 42

## Related PRs and Issues

x

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by lohanidamodar — APPROVED** [review]
Looks good to me. Let's just remember to update our docs regarding certificates about these changes, telling people to set the domain  env if they run into this issue and more.

File: app/controllers/general.php [hunk 1]
```diff
@@ -73,33 +73,44 @@
         } else {
             Authorization::disable();
 
-            $domainDocument = $dbForConsole->findOne('domains', [
-                new Query('domain', QUERY::TYPE_EQUAL, [$domain->get()])
-            ]);
+            $mainDomain = null;
+            if(!empty(App::getEnv('_APP_DOMAIN_TARGET', ''))) {
```
Comment (Meldiron) [inline app/controllers/general.php:77]: Domain target or app domain?

Comment (Meldiron) [reply app/controllers/general.php:77]: Target seems to be used for CNAME validation. Renaming to app_domain.

File: app/controllers/general.php [hunk 2]
```diff
@@ -73,33 +73,44 @@
...
+                        'registerable' => $domain->getRegisterable(),
+                        'verification' => false,
+                        'certificateId' => null,
+                    ]);
+    
+                    $domainDocument = $dbForConsole->createDocument('domains', $domainDocument);
+    
+                    Console::info('Issuing a TLS certificate for the main domain (' . $domain->get() . ') in a few seconds...');
+    
+                    Resque::enqueue('v1-certificates', 'CertificatesV1', [
```
Comment (christyjacob4) [inline app/controllers/general.php:104]:
```suggestion
                    Resque::enqueue(Event::CERTIFICATES_QUEUE_NAME, Event:: CERTIFICATES_CLASS_NAME, [
```

Comment (christyjacob4) [reply app/controllers/general.php:104]: You will also need to import the Event class.
This file does not contain it.

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
some minor changes

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**